rio_tiler/io/cogeo.py

```diff
@@ -520,6 +521,91 @@ def point(
 
         return point
 
+    def feature(
+        self,
+        shape: Dict,
+        dst_crs: Optional[CRS] = None,
+        shape_crs: CRS = constants.WGS84_CRS,
+        max_size: int = 1024,
+        indexes: Optional[Union[int, Sequence]] = None,
+        expression: Optional[str] = "",
+        **kwargs: Any,
+    ) -> ImageData:
+        """
+        Read a COG for a geojson feature.
+
+        Attributes
+        ----------
+        shape: dict
+            Valid GeoJSON feature.
+        dst_crs: CRS or str, optional
+            Target coordinate reference system, default is the bbox CRS.
+        shape_crs: CRS or str, optional
+            shape coordinate reference system, default is "epsg:4326"
+        max_size: int, optional
+            Limit output size array, default is 1024.
+        indexes: int or sequence of int
+            Band indexes (e.g. 1 or (1, 2, 3))
+        expression: str
+            rio-tiler expression (e.g. b1/b2+b3)
+        kwargs: dict, optional
+            These will be passed to the 'rio_tiler.reader.part' function.
+
+        Returns
+        -------
+        data: numpy ndarray
+        mask: numpy array
+
+        """
+        kwargs = {**self._kwargs, **kwargs}
+
+        if isinstance(indexes, int):
+            indexes = (indexes,)
+
+        if indexes and expression:
+            warnings.warn(
+                "Both expression and indexes passed; expression will overwrite indexes parameter.",
+                ExpressionMixingWarning,
+            )
+
+        if expression:
+            indexes = parse_expression(expression)
+
+        if not dst_crs:
+            dst_crs = shape_crs
+
+        # Get BBOX of the polygon
+        bbox = featureBounds(shape)
```

Comment:
Do you want to raise something like a `FeatureOutsideBoundsError` when the bbox doesn't intersect image coordinates?

Reply:
I don't think we raise error like that in `.part()` method so I feel it's ok to not raise anything 🤷‍♂️ . The fact we raise an error `TileOutsideBounds` within the `.tile()` method is first historical but also because it requires more knowledge to understand if a tile is outside or outside a dataset bounds. 

Again not 💯 sold on `not throwing error`, so if you have a good argument I'm listening 😄 !